### PR TITLE
No longer crash during `destroy()` of `MultiRootEditor` or `InlineEditor` when the editable was detached manually before destroying.

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Fix (editor-inline, editor-multi-root):  No longer crash the editor during `destroy()` when the editable was manually detached before destroying. Closes https://github.com/ckeditor/ckeditor5/issues/16561

---

### Additional information

Some editor types check if elements exist before destroying, while others don't. Here's what works correctly:

https://github.com/ckeditor/ckeditor5/blob/ck/16561/packages/ckeditor5-editor-classic/src/classiceditorui.ts#L141-L143
https://github.com/ckeditor/ckeditor5/blob/ck/16561/packages/ckeditor5-editor-balloon/src/ballooneditorui.ts#L102-L104
https://github.com/ckeditor/ckeditor5/blob/ck/16561/packages/ckeditor5-editor-decoupled/src/decouplededitorui.ts#L96-L98

These editors don't have the checks yet:

https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-editor-inline/src/inlineeditorui.ts#L117-L119
https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-editor-multi-root/src/multirooteditorui.ts#L162-L164

This PR adds missing checks to fix crashes in React integration shown here:

1. https://github.com/ckeditor/ckeditor5-react/blob/474c8aa9c222b61ed86dad329e05beb92ede29a8/src/useMultiRootEditor.tsx#L64-L88
2. https://github.com/ckeditor/ckeditor5/issues/16561

tl;dr React integration removes editables from DOM just before destroying the editor in Strict Mode. This was causing the multiroot editor to crash.

### CODE REVIEW

**Comment by pomek** [comment]
AFAIR, there was also a case where somebody called `#destroy()` twice, and it was also thrown. Could we check if before and after your changes it is valid? If so, I'd add one more test.

As for me, LGTM, but somebody from the Core team should accept it before merging. I am not too familiar with the editor's internals.

**Comment by Mati365** [comment]
Sure thing, I'll take a look.

**Comment by Mati365** [comment]
@pomek  

> AFAIR, there was also a case where somebody called #destroy() twice, and it was also thrown. Could we check if before and after your changes it is valid? If so, I'd add one more test.

Confirm, the exception was thrown in inline editor and multi root editor. So it looks like this one fix solved two independent React issues (that one with multiroots and the other in the builder).

**Review by niegowski — APPROVED** [review]
